clients/cli/src/main/java/org/apache/gravitino/cli/commands/ListColumns.java

```diff
@@ -59,10 +59,42 @@ public void handle() {
     } catch (NoSuchTableException noSuchTableException) {
       exitWithError(
           ErrorMessages.UNKNOWN_TABLE + Joiner.on(".").join(metalake, catalog, schema, table));
+      return;
     } catch (Exception exp) {
       exitWithError(exp.getMessage());
+      return;
     }
 
-    printResults(columns);
+    // Null / empty check before processing columns
+    if (columns == null || columns.length == 0) {
+      exitWithError("No columns found for table: " + table);
+      return;
+    }
+
+    StringBuilder all = new StringBuilder();
+    all.append("name,datatype,comment,nullable,auto_increment").append(System.lineSeparator());
```

Comment:
I believe we had a conversation over the "autoIncrement" field ...
It is not supposed appear on every column.